components/switch/test/switch.test.js

```diff
@@ -45,4 +45,29 @@ describe('d2l-switch', () => {
 		});
 	});
 
+	describe('consumer manages state', () => {
+
+		it('click with no state management', async() => {
+			const elem = await fixture(switchFixture);
+			const clickTarget = elem.shadowRoot.querySelector('.d2l-switch-container');
```

Comment:
I'm surprised you need a special click target... `await clickElem(elem)` should click right in the centre of the switch. Was that not working? Maybe because of the text to the side?

Reply:
Maybe. I will try it again. I was having some difficulty with these tests.

Reply:
If you do need a special target, it might be worth putting a `getClickElem()` method on the component itself to centralize that logic. I could see other consumers needing to do something similar in their test code, and having everyone reach into the shadow root isn't amazing.

Reply:
Ok, got that sorted. I think the missing change event was causing me some grief when I was doing the tests, and somewhere along the line I copied the click target code from elsewhere.

Reply:
(I updated to test to include an assertion for that event too)